pkg/controller/nodes/task/k8s/plugin_collector.go

```diff
@@ -0,0 +1,172 @@
+func (r *ResourceLevelMonitor) countList(ctx context.Context, objects []interface{}) map[string]int {
+	// Map of namespace to counts
+	counts := map[string]int{}
+
+	// Collect the object counts by namespace
+	for _, v := range objects {
+		metadata, err := meta.Accessor(v)
+		if err != nil {
+			logger.Errorf(ctx, "Error converting obj %v to an Accessor %s\n", v, err)
+			continue
+		}
+		counts[metadata.GetNamespace()]++
+	}
+
+	return counts
+}
+
+// The context here is expected to already have a value for the KindKey
+func (r *ResourceLevelMonitor) collect(ctx context.Context) {
+	// Emit gauges at the namespace layer - since these are just K8s resources, we cannot be guaranteed to have the necessary
+	// information to derive project/domain
+	objects := r.sharedInformer.GetStore().List()
+	counts := r.countList(ctx, objects)
+
+	for ns, count := range counts {
+		withNamespaceCtx := contextutils.WithNamespace(ctx, ns)
+		r.Levels.Set(withNamespaceCtx, float64(count))
+	}
+}
+
+func (r *ResourceLevelMonitor) RunCollector(ctx context.Context) {
+	ticker := time.NewTicker(resourceLevelMonitorCycleDuration)
+	collectorCtx := contextutils.WithGoroutineLabel(ctx, "k8s-resource-level-monitor")
+	// Since the idea is that one of these objects is always only responsible for exactly one type of K8s resource, we
+	// can safely set the context here for that kind for all downstream usage
+	collectorCtx = context.WithValue(ctx, KindKey, strings.ToLower(r.gvk.Kind))
+
+	go func() {
+		pprof.SetGoroutineLabels(collectorCtx)
+		r.sharedInformer.HasSynced()
+		logger.Infof(ctx, "K8s resource collector %s has synced", r.gvk.Kind)
+		for {
+			select {
+			case <-collectorCtx.Done():
+				return
+			case <-ticker.C:
+				t := r.CollectorTimer.Start(collectorCtx)
+				r.collect(collectorCtx)
+				t.Stop()
+			}
+		}
+	}()
+}
+
+func (r *ResourceLevelMonitor) RunCollectorOnce(ctx context.Context) {
+	r.once.Do(func() {
+		r.RunCollector(ctx)
+	})
+}
+
+// This struct is here to ensure that we do not create more than one of these monitors for a given GVK. It wouldn't necessarily break
+// anything, but it's a waste of compute cycles to compute counts multiple times. This can happen if multiple plugins create the same
+// underlying K8s resource type. If two plugins both created Pods (ie sidecar and container), without this we would launch two
+// ResourceLevelMonitor's, have two goroutines spinning, etc.
+type monitorIndex struct {
+	lock     *sync.Mutex
+	monitors map[schema.GroupVersionKind]*ResourceLevelMonitor
+}
+
+var index monitorIndex
+
+// These are declared here because this constructor will be called more than once, by different K8s resource types (Pods, SparkApps, OtherCrd, etc.)
+// and metric names have to be unique. It felt more reasonable at time of writing to have one metric and have each resource type just be a label
+// rather than one metric per type, but can revisit this down the road.
+var gauge *labeled.Gauge
```

Comment:
Why not initialize these here?
```suggestion
var gauge = labeled.NewGauge("k8s_resources", "Current levels of K8s objects as seen from their informer caches", scope, labeled.AdditionalLabelsOption{
		Labels: []string{contextutils.NamespaceKey.String(), KindKey.String()},
	})
```

And remove the more complex logic below?